ovs/lib/tc.c

```diff
@@ -2613,18 +2613,18 @@ nl_msg_put_flower_tunnel(struct ofpbuf *request, struct tc_flower *flower)
         nl_msg_put_in6_addr(request, TCA_FLOWER_KEY_ENC_IPV6_SRC, ipv6_src);
         nl_msg_put_in6_addr(request, TCA_FLOWER_KEY_ENC_IPV6_DST, ipv6_dst);
     }
-    if (tos_mask) {
+    if (tos || tos_mask) {
```

Comment:
IMO, getting approval for this change will be hard.  'tos' zero is valid and including these fields based on it's value will look odd. 

What we really want is to add all keys (irrespective of their value). Our requirement is to see the packet that triggered this flow creation. Getting people to see our requirement will be hard.